modules/tensor_mechanics/src/materials/cohesive_zone_model/BiLinearMixedModeTraction.C

```diff
@@ -16,237 +16,255 @@ InputParameters
+                                                "Critical energy release rate in shear direction.");
+  params.addRequiredParam<MaterialPropertyName>("normal_strength",
+                                                "Tensile strength in normal direction.");
+  params.addRequiredParam<MaterialPropertyName>("shear_strength",
+                                                "Tensile strength in shear direction.");
   params.addRequiredParam<Real>("eta", "The power law parameter.");
-  params.addParam<Real>("alpha", 1.0e-10, "The interpenetration regularization parameter.");
-  params.addParam<bool>(
-      "lag_seperation_state", false, "Lag seperation sate: use their old values.");
   MooseEnum criterion("POWER_LAW BK", "BK");
+  params.addParam<Real>("viscosity", 0.0, "Viscosity.");
   params.addParam<MooseEnum>(
       "mixed_mode_criterion", criterion, "Option for mixed mode propagation criterion.");
-  params.addCoupledVar("normal_strength_scale_factor", 1.0, "Scale factor for normal strength.");
-  params.addClassDescription("Mixed mode bilinear traction separation law.");
+
+  // Advanced parameters to improve numerical convergence
+  params.addParam<bool>(
+      "lag_mode_mixity", true, "Whether to use old displacement jumps to compute the mode mixity.");
+  params.addParam<bool>(
+      "lag_displacement_jump",
+      false,
+      "Whether to use old displacement jumps to compute the effective displacement jump.");
+  params.addParam<Real>("alpha", 1e-10, "Regularization parameter for the Macaulay bracket.");
+  params.addParamNamesToGroup("lag_mode_mixity lag_displacement_jump alpha", "Advanced");
+
   return params;
 }
 
 BiLinearMixedModeTraction::BiLinearMixedModeTraction(const InputParameters & parameters)
   : CZMComputeLocalTractionTotalBase(parameters),
-    _scale_factor(coupledValue("normal_strength_scale_factor")),
-    _stiffness(getParam<Real>("penalty_stiffness")),
+    _K(getParam<Real>("penalty_stiffness")),
     _d(declareProperty<Real>("damage")),
     _d_old(getMaterialPropertyOld<Real>("damage")),
     _interface_displacement_jump_old(
         getMaterialPropertyOld<RealVectorValue>("interface_displacement_jump")),
-    _viscosity(getParam<Real>("viscosity")),
-    _GI_C(getParam<Real>("GI_C")),
-    _GII_C(getParam<Real>("GII_C")),
-    _delta_normal0(getParam<Real>("normal_strength") / _stiffness),
-    _delta_shear0(getParam<Real>("shear_strength") / _stiffness),
-    _eff_disp_damage_init(
-        declareProperty<Real>(_base_name + "effective_displacement_at_damage_initiation")),
-    _eff_disp_damage_init_old(
-        getMaterialPropertyOld<Real>(_base_name + "effective_displacement_at_damage_initiation")),
-    _eff_disp_full_degradation(
-        declareProperty<Real>(_base_name + "effective_displacement_at_full_degradation")),
-    _eff_disp_full_degradation_old(
-        getMaterialPropertyOld<Real>(_base_name + "effective_displacement_at_full_degradation")),
```

Comment:
Those quantities are useful to be visualized in Paraview. Any thoughts?

Reply:
Yeah, fair. I can change them back to material properties.